spec/workers/sync_permissions_worker_spec.rb

```diff
@@ -7,10 +7,39 @@
     is_expected.to be_processed_in :user
   end
 
-  it "should sync user permissions" do
+  it "should call to sync user permissions" do
     user = create(:user)
     expect(User).to receive(:find_by_id).with(user.id).and_return(user)
     expect(user).to receive(:update_repo_permissions)
     subject.perform(user.id)
   end
+
+  context "with recorded data" do
+    before do
+      freeze_time
+    end
+
+    it "should sync user permissions and repositories" do
+      VCR.use_cassette("github/user_sync") do
+        user = create(:user)
+        user.identities.update_all(token: "TEST_TOKEN")
+
+        expect(user.repository_permissions.count).to be 0
+        expect(Repository.count).to be 0
+        expect(RepositoryDownloadWorker.jobs.size).to be 0
+
+        expect(User).to receive(:find_by_id).with(user.id).and_return(user)
```

Comment:
🎱 Can this be changed to the `have_received` expectations? I don't know the deets but iirc the `expect to receive` doesn't guarantee that the thing you want to have happen actually happens, probably similar to the sql `explain` vs `explain analyze` where one of them is the predicted path vs actually running the query to get the actual path.